app/assets/javascripts/app/cart.coffee

```diff
@@ -4,16 +4,28 @@ class window.CartQuantityReplacer
   toString: ->
     @originalPath.replace(/\bquantity=\d+(&|$)/, "quantity=#{@newQuantity}$1")
 
-$ ->
-  class Cart
-    constructor: (cartSelector) ->
-      @$cart = $(cartSelector)
-      @$cart.find("[data-quantity-field]").each @setupQuantityListeners
+class window.Cart
+  constructor: (cartSelector) ->
+    @$cart = $(cartSelector)
+    @$cart.find("[data-quantity-field]").each @setupQuantityListeners
+
+  setupQuantityListeners: (_i, link) ->
+    $link = $(link)
+    $quantityField = $("#" + $(link).data("quantity-field"))
+    originalHref = $link.attr("href")
+
+    $quantityField.change (e) ->
+      newQuantity = parseInt($quantityField.val())
 
-    setupQuantityListeners: (i, link) ->
-      trackedField = $(link).data("quantity-field")
-      $("##{trackedField}").change ->
-        newQuantity = $(@).val()
-        link.href = new CartQuantityReplacer(link.href, newQuantity)
+      if newQuantity > 0
+        $link
+          .removeClass("disabled")
+          .attr("href", new CartQuantityReplacer(originalHref, newQuantity))
 
+      else
+        $link
+          .addClass("disabled")
+          .removeAttr("href")
+
```

Comment:
I'm not sure. I feel like this is clearer than two things switching on the same condition. I'd be a little more comfortable using `toggleClass` if I could also address the attribute in a parallel structure with something like `.attr("href", quantity > 0 ? href : "")`, but an `<a>` without an `href` is different than an `<a href="">`.